library/camellia.c

```diff
@@ -345,6 +353,9 @@ int mbedtls_camellia_setkey_enc( mbedtls_camellia_context *ctx, const unsigned c
     uint32_t SIGMA[6][2];
     uint32_t KC[16];
     uint32_t TK[20];
+    CAMELLIA_VALIDATE_RET( ctx != NULL );
+    CAMELLIA_VALIDATE_RET( key != NULL );
+    CAMELLIA_VALIDATE_RET( keybits == 128 || keybits == 192 || keybits == 256 );
```

Comment:
This is already checked in the switch below. I would revert this particular check. However, this may require applying the deprecation scheme for the `..._INVALID_KEY_LENGTH` constant, which would, in turn, be generalized to a common error code.

Reply:
With the given implementation of this function, we cannot remove the unconditional check because an arbitrary value of `keybits` would corrupt the stack (the key is copied into the local stack buffer `t`).  While I think that in principle we should include the check `keybits` in the parameter validation regardless of the shape of the implementation (what's checked in parameter validation should be entirely determined by the function contract, and be independent of the implementation), in order to avoid the duplication, I'll remove it.